src/Security/Authorization/test/AuthorizationMiddlewareTests.cs

```diff
@@ -340,6 +341,33 @@ public async Task AuthZResourceShouldBeHttpContextAndHaveHEndpoint()
             Assert.Equal(endpoint, resource.GetEndpoint());
         }
 
+        [Fact]
+        public async Task AuthZResourceShouldBeEndpointByDefaultWithCompatSwitch()
+        {
+            AppContext.SetSwitch("Microsoft.AspNetCore.Authorization.SuppressUseHttpContextAsAuthorizationResource", isEnabled: true);
```

Comment:
I just realized you need to revert this at the end of the test or else you'll affect other tests.

Reply:
Are you sure?  This isn't something we do in any of our other tests that SetSwitch, otherwise we'd have to wrap the tests in try/finally right?

Reply:
AppContext affect the whole process. Any test run after this in the same process will have the quirked behavior.

The try/finally is only needed if the test fails.

Reply:
I ran into this on another PR today. I had tests in two classes that were affected by the value of SetSwitch, and those classes would run in parallel by default. I had to avoid calling SetSwitch in either of them in favor of setting an internal variable on a local instance.